bpf/bpf_lxc.c

```diff
@@ -1096,6 +1152,15 @@ int tail_ipv6_to_endpoint(struct __ctx_buff *ctx)
 	if (IS_ERR(ret))
 		return send_drop_notify(ctx, src_identity, SECLABEL, LXC_ID,
 					ret, CTX_ACT_DROP, METRIC_INGRESS);
+
+#ifdef ENABLE_CUSTOM_CALLS
+	if (!encode_custom_prog_meta(ctx, ret, src_identity)) {
```

Comment:
This code doesn't seem to handle the proxy case, but we know it could occur up on line 1150 with the hairpin (which calls redirect and passes back the return code).

Reply:
In contrast to the case above, I think for this case it's probably OK to use `CTX_ACT_TX` because it's in the "to_endpoint" variation which is only ever run on the TC egress qdisc on the way into a pod so we *must* redirect the packets to send them through the proxy. If we didn't actually redirect the packets here then we would just let the packets through to the pod without traversing the proxy. That said, for consistency maybe it'd be better to make the logic consistent between the two cases.

Reply:
Thank you, and agreed for the consistency. I'll update.